From 091f9634bfe3ec0f33d6612f402beb1118e06b86 Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Mon, 18 Feb 2019 14:16:53 +0100
Subject: [PATCH] PONRTSYS-3636: Revert "DRVLIB_SW-1478: DP address review
 comment"

This reverts commit 82ccdabb8dd6233729dd96654df49eb6e37b496d.

This commit broke OMCI when it is done for the second time.
---
 drivers/net/ethernet/lantiq/cqm/grx500/cbm.c         |  1 -
 drivers/net/ethernet/lantiq/datapath/datapath_api.c  |  2 +-
 drivers/net/ethernet/lantiq/datapath/datapath_proc.c | 10 +++++-----
 3 files changed, 6 insertions(+), 7 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/cqm/grx500/cbm.c b/drivers/net/ethernet/lantiq/cqm/grx500/cbm.c
index 09fa05a19733..c10c1fe63972 100644
--- a/drivers/net/ethernet/lantiq/cqm/grx500/cbm.c
+++ b/drivers/net/ethernet/lantiq/cqm/grx500/cbm.c
@@ -3019,7 +3019,6 @@ dp_port_alloc(
 	data->dp_port = port_id;
 	data->deq_port = (flags & DP_F_FAST_WLAN) ? cbm_port[i] : cbm_port[0];
 	data->deq_port_num = 1;
-	data->num_dma_chan = 1;
 	return CBM_SUCCESS;
 }
 
diff --git a/drivers/net/ethernet/lantiq/datapath/datapath_api.c b/drivers/net/ethernet/lantiq/datapath/datapath_api.c
index 78af8000346f..90b4a6c26d9a 100644
--- a/drivers/net/ethernet/lantiq/datapath/datapath_api.c
+++ b/drivers/net/ethernet/lantiq/datapath/datapath_api.c
@@ -47,7 +47,7 @@ struct dma_rx_desc_1 dma_tx_desc_mask1;
 u32 dp_drop_all_tcp_err;
 u32 dp_pkt_size_check;
 
-u32 dp_dbg_flag=DP_DBG_FLAG_DBG|DP_DBG_FLAG_REG;
+u32 dp_dbg_flag;
 EXPORT_SYMBOL(dp_dbg_flag);
 
 #ifdef CONFIG_LTQ_DATAPATH_MPE_FASTHOOK_TEST
diff --git a/drivers/net/ethernet/lantiq/datapath/datapath_proc.c b/drivers/net/ethernet/lantiq/datapath/datapath_proc.c
index 7f536a45b663..44a30f7e4273 100644
--- a/drivers/net/ethernet/lantiq/datapath/datapath_proc.c
+++ b/drivers/net/ethernet/lantiq/datapath/datapath_proc.c
@@ -206,11 +206,11 @@ int proc_port_dump(struct seq_file *s, int pos)
 		cid = _DMA_CONTROLLER(dp_deq_port_tbl[tmp_inst][cqm_p].dma_chan);
 		pid = _DMA_PORT(dp_deq_port_tbl[tmp_inst][cqm_p].dma_chan);
 		nid = _DMA_CHANNEL(dp_deq_port_tbl[tmp_inst][cqm_p].dma_chan);
-		if (port->num_dma_chan)
-			seq_printf(s, "          : dma_ch/port:    %d/%d(ref=%d)\n",
-				   nid, cqm_p,
-				   atomic_read(&dp_dma_chan_tbl[tmp_inst][cid]
-						[pid][nid].ref_cnt));
+		seq_printf(s, "          : dma_ch/node:    0x%x/%d(ref=%d)\n",
+			   nid,
+			   port->subif_info[i].qos_deq_port,
+			   atomic_read(&dp_dma_chan_tbl[tmp_inst][cid][pid]
+					[nid].ref_cnt));
 
 		if (port->subif_info[i].ctp_dev &&
 		    port->subif_info[i].ctp_dev->name)
